tsdb/index.go

```diff
@@ -2918,12 +2933,18 @@ func (is IndexSet) tagValuesByKeyAndExpr(auth query.FineAuthorizer, name []byte,
 
 		if auth != nil {
 			name, tags := ParseSeriesKey(buf)
+			if name == nil && tags == nil {
```

Comment:
Would it be enough to bail if `name == nil`?

Reply:
Agreed!

Similar to what we discussed yesterday, there is a similar prior check above this part so it _should_ be caught already. A `name == nil` on the parsing is a definite invalid value we want to skip.

Just above the highlighted code snippet for this comment :point_down: 
```go
		buf := is.SeriesFile.SeriesKey(e.SeriesID)
		if len(buf) == 0 {
			continue
		}
```